### PR TITLE
update nodejs to v22.11.0

### PR DESCRIPTION
**Pull request by anonrig**
Update Node.js to latest version to support features like `Promise.withResolvers()`

### CODE REVIEW

File: build/deps/build_deps.jsonc [hunk 1]
```diff
@@ -93,6 +93,15 @@
...
       "file_regex": "\\.tar\\.gz$"
     },
+    {
+      // Needed for running the latest Node.js due to aspect_rules_js
+      "name": "rules_nodejs",
+      "type": "github_release",
+      "owner": "bazel-contrib",
+      "repo": "rules_nodejs",
+      "file_regex": "\\.tar\\.gz$",
+      "freeze_version": "v6.3.2"
```
Comment (fhanau) [inline build/deps/build_deps.jsonc:103]: - [ ] No need to freeze this – later versions will work just as well

File: build/deps/build_deps.jsonc [hunk 2]
```diff
@@ -93,6 +93,15 @@
       "repo": "rules_pkg",
       "file_regex": "\\.tar\\.gz$"
     },
+    {
+      // Needed for running the latest Node.js due to aspect_rules_js
```
Comment (fhanau) [inline build/deps/build_deps.jsonc:97]: Nit: rules_js is not making the situation worse per se – we'd pull in an outdated version using `rules_js_dependencies()` without this change, but if we weren't using rules_js we'd still have to import it ourselves. This comment should say something like "due to aspect_rules_js using older version by default".

**Review by fhanau — APPROVED** [review]

File: build/deps/build_deps.jsonc (hunk 2 above)
Comment (anonrig) [reply build/deps/build_deps.jsonc]: Thanks! I updated the coment.

File: build/deps/build_deps.jsonc (hunk 1 above)
Comment (anonrig) [reply build/deps/build_deps.jsonc]: Thanks! Removed it.

**Pull request merged**